vNext/AISKU/Tests/applicationinsights.e2e.tests.ts

```diff
@@ -658,7 +665,33 @@ export class ApplicationInsightsTests extends TestClass {
                 currentCount += call[1];
             });
             console.log('curr: ' + currentCount + ' exp: ' + expectedCount, ' appId: ' + this._ai.context.appId());
-            return currentCount === expectedCount && !!this._ai.context.appId();
+            let isValid: boolean = false;
+            if (currentCount === expectedCount) {
+                const payloadStr: string[] = this.successSpy.args[0][0];
+                const payload = JSON.parse(payloadStr[0]);
+                const baseType = payload.data.baseType;
+                // call the appropriate Validate depending on the baseType
+                switch (baseType) {
+                    case Event.dataType:
+                        return EventValidator.EventValidator.Validate(payload, baseType);
+                    case Trace.dataType:
+                        return TraceValidator.TraceValidator.Validate(payload, baseType);
+                    case Exception.dataType:
+                        return ExceptionValidator.ExceptionValidator.Validate(payload, baseType);
+                    case Metric.dataType:
+                        return MetricValidator.MetricValidator.Validate(payload, baseType);
+                    case PageView.dataType:
+                        return PageViewValidator.PageViewValidator.Validate(payload, baseType);
+                    case PageViewPerformance.dataType:
+                        return PageViewPerformanceValidator.PageViewPerformanceValidator.Validate(payload, baseType);
+                    case RemoteDependencyData.dataType:
+                        return RemoteDepdencyValidator.RemoteDepdencyValidator.Validate(payload, baseType);
+        
+                    default:
+                        return EventValidator.EventValidator.Validate(payload, baseType);
+                }
+            }
+            return currentCount === expectedCount && isValid && !!this._ai.context.appId();
```

Comment:
It does not look like this `return` will ever get reached if `currentCount === expectedCount`

Reply:
yes you are right.. fixed it! 👍